polus-precompute-slide-plugin/src/utils.py

```diff
@@ -97,65 +237,101 @@ def _get_higher_res(S,bfio_reader,slide_writer,encoder,X=None,Y=None):
         X = [0,scale_info['size'][0]]
     if Y == None:
         Y = [0,scale_info['size'][1]]
+    Z = [0,1]
     
     # Modify upper bound to stay within resolution dimensions
     if X[1] > scale_info['size'][0]:
         X[1] = scale_info['size'][0]
     if Y[1] > scale_info['size'][1]:
         Y[1] = scale_info['size'][1]
-    
+
+    datatype = bfio_reader.read_metadata().image().Pixels.get_PixelType()
+
     # Initialize the output
-    image = np.zeros((Y[1]-Y[0],X[1]-X[0]),dtype=bfio_reader.read_metadata().image().Pixels.get_PixelType())
+    image = np.zeros((Y[1]-Y[0],X[1]-X[0],1),dtype=datatype)
     
-    # If requesting from the lowest scale, then just read the image
     if str(S)==encoder.info['scales'][0]['key']:
-        image = bfio_reader.read_image(X=X,Y=Y).squeeze()
+        if slices == None:
+            image = bfio_reader.read_image(X=X,Y=Y,Z=Z)[...,0,0] 
+        else:
+            image = bfio_reader.read_image(X=X,Y=Y,Z=slices)[...,0,0] 
```

Comment:
Why not just have `Z` as an input value? The if statement appears to unnecessary, and it looks like it might even create an issue when you want to have a single slice segmentation.